main/tests/cypress/cypress/integration/project/grid/column/facet/facets.spec.js

```diff
@@ -1,184 +1,417 @@
-    })
+	it('Test the facets panel (no facets yet)', function () {
+		cy.loadAndVisitProject('food.small')
+		cy.columnActionClick('Water', ['Facet', 'Text facet'])
+		cy.get('#refine-tabs-facets')
+			.should('exist')
+			.contains('Using facets and filters')
+	})
+
+	it('Test the display of multiple facets', function () {
+		cy.loadAndVisitProject('food.small')
+		cy.columnActionClick('NDB_No', ['Facet', 'Text facet'])
+		cy.columnActionClick('Shrt_Desc', ['Facet', 'Text facet'])
+		cy.columnActionClick('Water', ['Facet', 'Text facet'])
+
+		cy.getFacetContainer('NDB_No').should('exist')
+		cy.getFacetContainer('Shrt_Desc').should('exist')
+		cy.getFacetContainer('Water').should('exist')
+	})
+
+	it('Test the Remove All button', function () {
+		cy.loadAndVisitProject('food.small')
+		cy.columnActionClick('NDB_No', ['Facet', 'Text facet'])
+		cy.columnActionClick('Shrt_Desc', ['Facet', 'Text facet'])
+		cy.get('#refine-tabs-facets .facets-container .facet-container').should(
+			'have.length',
+			2
+		)
+
+		cy.get('#refine-tabs-facets a').contains('Remove All').click()
+		cy.get('#refine-tabs-facets .facets-container .facet-container').should(
+			'have.length',
+			0
+		)
+		cy.get('#refine-tabs-facets')
+			.should('exist')
+			.contains('Using facets and filters')
+	})
+
+	it('Test the Reset All button', function () {
+		cy.loadAndVisitProject('food.small')
+		cy.columnActionClick('Water', ['Facet', 'Text facet'])
+		cy.columnActionClick('Energ_Kcal', ['Facet', 'Text facet'])
+		cy.get('#refine-tabs-facets .facets-container .facet-container').should(
+			'have.length',
+			2
+		)
+
+		// Click to facet 1, select first value
+		cy.getFacetContainer('Water')
+			.find('.facet-choice:first-child .facet-choice-label')
+			.click()
+		cy.getFacetContainer('Water')
+			.find('.facet-choice:first-child')
+			.should('have.class', 'facet-choice-selected')
+
+		// Click to facet 2, select first value
+		cy.getFacetContainer('Energ_Kcal')
+			.find('.facet-choice:first-child .facet-choice-label')
+			.click()
+		cy.getFacetContainer('Energ_Kcal')
+			.find('.facet-choice:first-child')
+			.should('have.class', 'facet-choice-selected')
+
+		cy.get('#refine-tabs-facets a').contains('Reset All').click()
+
+		// all facets selections should be gone
+		cy.getFacetContainer('Water')
+			.find('.facet-choice:first-child')
+			.should('not.have.class', 'facet-choice-selected')
+		cy.getFacetContainer('Energ_Kcal')
+			.find('.facet-choice:first-child')
+			.should('not.have.class', 'facet-choice-selected')
+	})
+
+	it('Create a simple text facet, check various elements in a Facet', function () {
+		cy.loadAndVisitProject('food.small')
+		cy.columnActionClick('Water', ['Facet', 'Text facet'])
+		cy.get('#refine-tabs-facets').should('exist')
+
+		cy.getFacetContainer('Water').should('exist')
+		cy.getFacetContainer('Water').contains('182 choices')
+		cy.getFacetContainer('Water').contains('Sort by')
+		cy.getFacetContainer('Water').contains('Cluster')
```

Comment:
@thadguidry this syntax will not work, because `get` only accepts a selector.
Moreover, `cy.get` always starts from the entire document, [explained here](https://docs.cypress.io/api/commands/get.html#Get-in-within) 
You could use `within`, but actually if you just want to refine the selector and ensure it's a button, you could simply replace the line by `cy.getFacetContainer('Water').find('button').contains('Cluster')` (tested and working)

Reply:
Ah, OK.